### PR TITLE
fix: Tabs fn should be overrideable

### PR DESCRIPTION
**Pull request by scomea**
## 📖 Description

The protected "setTabs" function in the `Tabs` component was written as an arrow function which made overriding it difficult as you can't simply call `super.setTabs()` to invoke the base class function when extending the `Tabs` class.  This change converts that function to a class member so it can be overridden more easily.  Additional minor cleanup of component code.

This is a breaking change in that anyone who currently has a workaround to override this function will need to make changes (ie. storing the base class arrow function and invoking it, for example).

### 🎫 Issues
Ad hoc.

### General
- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific
- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Review by KingOfTac — APPROVED** [review]
Thanks Stephane!

File: packages/web-components/fast-foundation/src/tabs/tabs.ts [hunk 1]
```diff
@@ -145,6 +145,11 @@ export class FASTTabs extends FASTElement {
         }
     }
 
+    /**
+     * Function that is invoked whenever the selected tab or the tab collection changes.
+     *
+     * @protected
```
Comment (radium-v) [inline packages/web-components/fast-foundation/src/tabs/tabs.ts:151]: This tsdoc tag indicates whether the method is in the public API or is internal, and isn't connected to the TypeScript scope.

```suggestion
     * @public
```

Comment (scomea) [reply packages/web-components/fast-foundation/src/tabs/tabs.ts]: done

**Review by bheston — APPROVED** [review]
It looks like that was the convention for many of the private functions. Is there a benefit to this model in any case? I wonder if we should document it as a coding convention for any particular use.

**Comment by chrisdholt** [comment]
> It looks like that was the convention for many of the private functions. Is there a benefit to this model in any case? I wonder if we should document it as a coding convention for any particular use.

The intent of "protected" is to enable extensibility for classes which extend this as a base class while maintaining their behavior (especially those which are core to the functionality or something like accessibility). https://www.typescriptlang.org/docs/handbook/2/classes.html#member-visibility

**Comment by bheston** [comment]
> ...intent of "protected"...

I was referring to the arrow function mechanism. I'm not sure what the benefit of that style is as it doesn't support the normal override capability. I was reading it as something you could _call_ from a subclass but not override, like a sort of hybrid private/protected. It seemed intentional though.

**Comment by chrisdholt** [comment]
> > ...intent of "protected"...
> 
> I was referring to the arrow function mechanism. I'm not sure what the benefit of that style is as it doesn't support the normal override capability. I was reading it as something you could _call_ from a subclass but not override, like a sort of hybrid private/protected. It seemed intentional though.

Gotcha - thx for clarification here! We often use arrows for events, we should probably avoid for protected classes where we do setup like this. I think that’s a good call out.

File: change/@microsoft-fast-foundation-879984ab-c986-4ae7-93f2-bbebf4530962.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "tabs fn cleanup",
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-879984ab-c986-4ae7-93f2-bbebf4530962.json:3]:
```suggestion
  "comment": "fix: allow tabs `setTabs` method to be extended",
```

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**